src/main/java/swinglib/JLabelBuilder.java

```diff
@@ -37,15 +41,25 @@ public static JLabelBuilder builder() {
 
   /**
    * Constructs a Swing JLabel using current builder values.
-   * Values that must be set: text
+   * Values that must be set: text or icon
    */
   public JLabel build() {
-    Preconditions.checkNotNull(text);
-    Preconditions.checkState(!text.trim().isEmpty());
+    Preconditions.checkState(text != null || icon != null);
```

Comment:
A second precondition could be missing, a check that one of text or icon is null.eg:

```
  Preconditions.checkState(text != null || icon != null);
  Preconditions.checkState(text == null || icon == null);
```
Really an 'xor', could simplify to one even.

Point though is that choosing both icon and text would be a configuration error, indeterministic, so we can fail in such a case.

Reply:
Why would we disallow a label with both an icon and text?  `JLabel` permits this configuration.  Do we want to add that additional constraint?

Actually, looking back on how I implemented the icon in the error dialog, I probably should have just added the icon to the existing `errorMessage` label instead of creating a separate label for the icon.

Reply:
Ah, I did not know `JLabel` would support both, That changes my point of view, I agree `or` is appropriate over `xor`